testsuite/MDAnalysisTests/analysis/test_bat.py

```diff
@@ -127,3 +127,12 @@ def test_bat_incorrect_dims(self, bat_npz):
         errmsg = 'Dimensions of array in loaded file'
         with pytest.raises(ValueError, match=errmsg):
             R = BAT(selected_residues, filename=bat_npz)
+
+    def test_bat_transformation(self, selected_residues, bat):
+        R = BAT(selected_residues)
+        XYZ = R.Cartesian(bat[0])
```

Comment:
You don't use this line.

Reply:
Umm actually  , the Cartesian method should atleast run twice before the co-ordinates are checked with the original ones , which is why I think it's necessary to put that line . 😅

Reply:
In that case, my mistake -- this test is not actually testing the right problem. I think you are testing an indirect side effect by comparing the Cartesian transformation. It would be better to directly test the problem noticed in https://github.com/MDAnalysis/mdanalysis/issues/3501#issue-1098151358 by checking that `bat[0]` has not changed. Otherwise this test is quite confusing as written.

Reply:
Umm actually the issue arises while we run the cartesian method  in torsions = bat_frame[2*n_torsions + 9:] due to pass by reference. But as the cartesian method doesn't return torsions how do i test that ?
The change in torsions results in a change in cartesian co-ords which are returned by the method which is why i tested that .

Reply:
Which is why i am checking the cartesian-cords after the 2nd transformation (PQR) and after the 1st transformation(XYZ)

Reply:
@umak1106 if I understand the issue correctly, the problem is not that PQR != XYZ, but that `bat[0]` before executing `R.Cartesian` does not equal `bat[0]` afterwards.

Reply:
Yes mam , but i tried writing this before and comparing bat[0] but the test doesn't fail without the fix and there is no way of
accessing bat_frame/torsions from the cartesian method either so am kinda stuck . Could you suggest what i should do instead ?
R = BAT(selected_residues)
        R.run()
        R.Cartesian(bat[0])
        XYZ = R.results.bat[0]
        R.Cartesian(bat[0])
        PQR = R.results.bat[0]
        assert_almost_equal(XYZ, PQR, 5,
                            err_msg="error: Reconstructed Bat"
                                    "coordinates after multiple "
                                    "transformations don't match"
                                    "original")

Reply:
Is there any other `bat` in the test?

Reply:
No I don't think so 😅.

Reply:
What do you pass to R.Cartesian?

Reply:
Bat-frame

Reply:
Yup I figured out my mistake . Ill make the changes right now in 5 mins